content/300-guides/050-database/890-neon.mdx

```diff
@@ -0,0 +1,154 @@
+
+```text
+DATABASE_URL=postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb?connect_timeout=10`
+```
+
+<Admonition type="note">
+
+A `connect_timeout` setting of 0 means no timeout.
+
+</Admonition>
+
+Another possible cause of connection timeouts is [Prisma's connection pool](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/), which has a default timeout of 10 seconds. This is typically enough time for Neon, but if you are still experiencing connection timeouts, you can try increasing this limit (in addition to the `connect_timeout` setting described above) by setting the `pool_timeout` parameter to a higher value. For example:
+
+```text
+DATABASE_URL=postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb?connect_timeout=15&pool_timeout=15`
+```
+
+## How to use Neon's serverless driver with Prisma (Preview)
+
+The [Neon serverless driver](https://github.com/neondatabase/serverless) is a low-latency Postgres driver for JavaScript and TypeScript that allows you to query data from serverless and edge environments over HTTP or WebSockets in place of TCP. Prisma by default uses TCP.
+
+You can use Prisma along with the Neon serverless driver using a [driver adapter](/concepts/components/drivers) <span class="concept"></span>. A driver adapter allows you to use a different database driver, from the default Prisma provides, to communicate with your database.
+
+<Admonition>
+
+This feature is available from Prisma versions 5.4.2 and later.
+
+</Admonition>
+
+To get started, enable the `driverAdapters` Preview feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+```
+
+Generate Prisma Client:
+
+```sh
+npx prisma generate
+```
+
+Install the Prisma adapter for Neon, Neon serverless driver and ws packages:
+
+```sh
+npm install @prisma/adapter-neon @neondatabase/serverless ws
+```
+
+Update your Prisma Client instance:
+
+```ts
+import { Pool, neonConfig } from '@neondatabase/serverless'
+import { PrismaNeon } from '@prisma/adapter-neon'
+import { PrismaClient } from '@prisma/client'
+import dotenv from 'dotenv'
+import ws from 'ws'
+
+dotenv.config()
+neonConfig.webSocketConstructor = ws
+const connectionString = `${process.env.DATABASE_URL}`
+
+const pool = new Pool({ connectionString })
+const adapter = new PrismaNeon(pool)
+const prisma = new PrismaClient({ adapter })
+```
+
+You can then use Prisma Client as you normally would with full-type-safety.
```

Comment:
This code block is now wrapped 🎉

Reply:
The Serverless driver adapter is a Prisma Client-specific concept. Therefore, the `DIRECT_URL`/ `DATABASE_URL` configured in the Prisma schema doesn't seem to affect Introspection and Migration.

Could you clarify what you mean or provide a suggestion to clarify what you mean?

Reply:
I am not using the driver adapter.
How do I run migrations and introspection?

Reply:
Migration and introspection would still work with/ without the driver adapter. There is no extra configuration needed. 😄

Reply:
Driver Adapters does not need the `DATABASE_URL` in your schema, you can just put a fake value
How will Migration and Introspection work then?

Reply:
I got clarification from Mahmoud that you can use either the pooled or the non-pooled connection string with the Neon database driver.

I'll add the note for the user that they should ensure that what they set up would still work with Prisma Migrate. I'm not sure what this would look like but I'll draft something up for feedback.

Reply:
I've added a datasource block with a `DATABASE_URL` in the segment enabling the Preview feature instead to match what is used as the connection string in the driver adapter snippet.